test/run

```diff
@@ -0,0 +1,258 @@
+        result=0
+      fi
+      break
+    fi
+    attempt=$(( $attempt + 1 ))
+    sleep $sleep_time
+  done
+  return $result
+}
+
+scl_usage() {
+  local run_cmd="$1"
+  local expected="$2"
+
+  info "Testing the image SCL enable"
+  out=$(docker run --rm ${IMAGE_NAME} /bin/bash -c "${run_cmd}")
+  if ! echo "${out}" | grep -q "${expected}"; then
+    echo "ERROR[/bin/bash -c "${run_cmd}"] Expected '${expected}', got '${out}'"
+    return 1
+  fi
+  out=$(docker exec $(cat ${cid_file}) /bin/bash -c "${run_cmd}" 2>&1)
+  if ! echo "${out}" | grep -q "${expected}"; then
+    echo "ERROR[exec /bin/bash -c "${run_cmd}"] Expected '${expected}', got '${out}'"
+    return 1
+  fi
+  out=$(docker exec $(cat ${cid_file}) /bin/sh -ic "${run_cmd}" 2>&1)
+  if ! echo "${out}" | grep -q "${expected}"; then
+    echo "ERROR[exec /bin/sh -ic "${run_cmd}"] Expected '${expected}', got '${out}'"
+    return 1
+  fi
+  return 0
+}
+
+function test_scl_usage() {
+  scl_usage "ruby --version" "ruby ${RUBY_VERSION}."
+  check_result "$?"
+}
+
+function test_npm_functionality() {
+  info "Testing npm availibility"
+  ct_npm_works
+  check_result "$?"
+}
+
+function run_all_tests() {
+  local APP_NAME="$1"
+  for test_case in $TEST_SET; do
+    info "Running test $test_case ... "
+    TESTCASE_RESULT=0
+    $test_case
+    check_result $?
+    local test_msg
+    if [ $TESTCASE_RESULT -eq 0 ]; then
+      test_msg="[PASSED]"
+    else
+      test_msg="[FAILED]"
+      TESTSUITE_RESULT=1
+    fi
+    printf -v test_short_summary "%s %s for '%s' %s\n" "${test_short_summary}" "${test_msg}" "${APP_NAME}" "$test_case"
+    [ -n "${FAIL_QUICKLY:-}" ] && {
+      cleanup "${APP_NAME}"
+      return 1
+    }
+  done;
+
+  cleanup "${APP_NAME}"
+}
+
+function test_application() {
+  # Verify that the HTTP connection can be established to test application container
+  run_test_application &
+
+  # Wait for the container to write it's CID file
+  wait_for_cid
+  check_result $?
+}
+
+function test_scl_variables_in_dockerfile() {
+  if [ "$OS" == "rhel7" ] || [ "$OS" == "centos7" ]; then
+    # autocleanup only enabled here as only the following tests so far use it
+    CID_FILE_DIR=$(mktemp -d)
+    ct_enable_cleanup
+
+    info "Testing variable presence during \`docker exec\`"
+    ct_check_exec_env_vars
+    check_result $?
+
+    info "Checking if all scl variables are defined in Dockerfile"
+    ct_check_scl_enable_vars
+    check_result $?
+ fi
+}
+
+function test_from_dockerfile() {
+  rootdir=$(readlink -zf ${image_dir}/..)
+  dockerfile="Dockerfile${1:-}"
+  info "Check building using a $dockerfile"
+  ct_test_app_dockerfile $rootdir/examples/from-dockerfile/$dockerfile 'https://github.com/sclorg/rails-ex.git' 'Welcome to your Rails application on OpenShift' app-src
+  check_result $?
+}
+
+pushd ${test_dir}
+if [ -d db-test-app ]; then
+  rm -rf db-test-app
+fi
+git clone git://github.com/openshift/ruby-hello-world db-test-app
+popd
+
+test_from_dockerfile
```

Comment:
This is not mentioned in the result text. Please fix it.

Reply:
All tests passed but `test_from_dockerfile` is not mentioned.
```
[PASSED] for '\''db'\'' test_docker_run_usage
[PASSED] for '\''db'\'' test_application
[PASSED] for '\''db'\'' test_connection
[PASSED] for '\''db'\'' test_scl_variables_in_dockerfile
[PASSED] for '\''db'\'' test_scl_usage
[PASSED] for '\''db'\'' test_npm_functionality
[PASSED] for '\''puma'\'' test_docker_run_usage
[PASSED] for '\''puma'\'' test_application
[PASSED] for '\''puma'\'' test_connection
[PASSED] for '\''puma'\'' test_scl_variables_in_dockerfile
[PASSED] for '\''puma'\'' test_scl_usage
.. snip ..
[PASSED] for 'rack' test_scl_variables_in_dockerfile
[PASSED] for 'rack' test_scl_usage
[PASSED] for 'rack' test_npm_functionality
```

Reply:
> This is not mentioned in the result text. Please fix it.

Done, let's see whether it helps.

[test] [test-openshift]

Reply:
This fix works for me.